Static Linux SDK: build errors at cmake build

default access levels after clone in subdir of $HOME:
```
xaver@xaver-HP-EliteBook-860-16-inch-G10-Notebook-PC:~/code/static-linux-test/swift-ci/sdks/static-linux$ ll
total 36
drwxrwxr-x 5 xaver xaver 4096 Apr  5 14:52 ./
drwxrwxr-x 3 xaver xaver 4096 Apr  5 14:52 ../
-rwxrwxr-x 1 xaver xaver 1266 Apr  5 14:52 build*
-rw-rw-r-- 1 xaver xaver 2616 Apr  5 14:52 Dockerfile
-rw-rw-r-- 1 xaver xaver   16 Apr  5 14:52 .gitignore
-rw-rw-r-- 1 xaver xaver  898 Apr  5 14:52 README.md
drwxrwxr-x 5 xaver xaver 4096 Apr  5 14:52 resources/
drwxrwxr-x 2 xaver xaver 4096 Apr  5 14:52 scripts/
```
Executing: ./build

results in the following error at cmake build due to missing write access rights of ./source directory inside the docker container:
```
-- Check for working C compiler: /usr/bin/cc - broken
CMake Error at /usr/share/cmake-3.22/Modules/CMakeTestCCompiler.cmake:65 (file):
  file failed to open for writing (No such file or directory):

    /source/swift-project/cmake/build/CMakeFiles/CMakeError.log
Call Stack (most recent call first):
  CMakeLists.txt:12 (project)


CMake Error at /usr/share/cmake-3.22/Modules/CMakeTestCCompiler.cmake:69 (message):
  The C compiler

    "/usr/bin/cc"

  is not able to compile a simple test program.

  It fails with the following output:





  CMake will not be able to correctly generate this project.
Call Stack (most recent call first):
  CMakeLists.txt:12 (project)


-- Configuring incomplete, errors occurred!
CMake Error: Cannot open file for write: /source/swift-project/cmake/build/CMakeCache.txt.tmp83767
CMake Error: : System Error: No such file or directory
CMake Error: Unable to open cache file for save. /source/swift-project/cmake/build/CMakeCache.txt
CMake Error: : System Error: No such file or directory
```

--- a/swift-ci/sdks/static-linux/scripts/build.sh
+++ b/swift-ci/sdks/static-linux/scripts/build.sh
@@ -243,15 +243,15 @@ header "Building CMake from source"
 
 quiet_pushd ${source_dir}/swift-project/cmake
 run cmake -G 'Ninja' ./ \
-    -B ${source_dir}/swift-project/cmake/build \
-    -DCMAKE_INSTALL_PREFIX=${source_dir}/swift-project/cmake/install \
+    -B ${build_dir}/cmake/build \
+    -DCMAKE_INSTALL_PREFIX=${build_dir}/cmake/install \
     -DCMAKE_BUILD_TYPE=Release \
     -DCMAKE_USE_OPENSSL=OFF \
     -DBUILD_CursesDialog=OFF \
     -DBUILD_TESTING=OFF
-run ninja -C ${source_dir}/swift-project/cmake/build
-run ninja -C ${source_dir}/swift-project/cmake/build install
-run export PATH="${source_dir}/swift-project/cmake/install/bin:$PATH"
+run ninja -C ${build_dir}/cmake/build
+run ninja -C ${build_dir}/cmake/build install
+run export PATH="${build_dir}/cmake/install/bin:$PATH"
 quiet_popd
 run cmake --version
 

